### PR TITLE
Feature auto link

### LINKED ISSUE #185: Dragging of images doesn't work when inserting between nodes
trimmed

This gif shows if you try to drag a file between the nodes (when the horizontal black line shows), the image is not inserted but if you insert the image inline it works. 

As far as I understand a wrapper node needs to be created so that the image can be inserted into this. We can look into how prosemirror handles within the document image drag.

### LINKED ISSUE #186: Convert URL's to links
When a user types a url ``http:\\fred.com`` or ``www.fred.com`` change into a link.

### PR DESCRIPTION
**Pull request by lucywang000**
Fix #186

### CODE REVIEW

File: core/components/link.js [hunk 1]
```diff
@@ -111,6 +113,32 @@ function pluginsFactory() {
...
+  /(^|\s)(((http|https|ftp):\/\/)?(?:[^\s.:\/]+\.)+(?:com|net|io))\s$/;
+
+function autoLinkInputRule(type) {
+  return new InputRule(URL_REGEX, (state, match, start, end) => {
+    if (!match[0]) {
+      return null;
+    }
+    const [_, leadingSpace, text, scheme] = match;
+    // If no scheme, use default scheme http://
+    const href = scheme ? text : `http://${text}`;
```
Comment (kepta) [inline core/components/link.js:128]: Should we make it default to https?

**Review by kepta — APPROVED** [review]
Looks great !

File: core/components/link.js (hunk 1 above)
Comment (lucywang000) [reply core/components/link.js:149]: There are quite some website that are http only (e.g. corporate-internal websites). And for most https sites they would redirect http requests to https anyway.

**Comment by lucywang000** [comment]
@kepta I just updated the tld regexp, also handled a corner case (see the last commit)

**Pull request merged**